### ISSUE TITLE
add documentation about the return value of addCallbacks and friends

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5399]
|[<img alt="zooko's avatar" src="https://avatars.githubusercontent.com/u/467900?s=50" width="50" height="50">](https://github.com/zooko)| @zooko reported|
|-|-|
|Trac ID|trac#5399|
|Type|enhancement|
|Created|2011-11-19 14:05:29Z|
|Branch|https://github.com/twisted/twisted/tree/addcallbacks-ret-5399|

I needed to know this and I ended up having to read the source code to learn it. This patch will let the next person learn it from docstrings.

```
#!diff
=== modified file 'twisted/internet/defer.py'
--- twisted/internet/defer.py   2011-10-04 18:09:00 +0000
+++ twisted/internet/defer.py   2011-11-19 14:00:55 +0000
@@ -276,6 +276,8 @@
         Add a pair of callbacks (success and error) to this L{Deferred}.
 
         These will be executed when the 'master' callback is run.
+
+        Returns self.
         """
         assert callable(callback)
         assert errback == None or callable(errback)
@@ -341,6 +343,8 @@
         being returned from a callback, it is possible to cause the call
         stack size limit to be exceeded by chaining many L{Deferred}s
         together with C{chainDeferred}.
+
+        Returns self.
         """
         d._chainedTo = self
         return self.addCallbacks(d.callback, d.errback)

```

(I'll also attach this patch as a unified diff file.)

Attachments:

* [doc-retval-of-addCallbacks.diff.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3b8/3b8869e35908ee9687690d83c9d2cf05306ddf50/477296f26c0839cc5268cbb71083d8c9d4f18b73.txt) (782 bytes) - added by zooko on 2011-11-19 14:05:47Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5399 5399
type__enhancement enhancement
reporter__zooko zooko
priority__normal normal
milestone__ 
branch__branches_addcallbacks_ret_5399 branches/addcallbacks-ret-5399
branch_author__zooko zooko
status__closed closed
resolution__fixed fixed
component__core core
keywords__doc doc
time__1321711529000000 1321711529000000
changetime__1322960481000000 1322960481000000
version__None None
owner__jesstess jesstess
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [33277]) Branching to 'addcallbacks-ret-5399'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess commented|
|-|-|

(In [33278]) Apply doc-retval-of-addCallbacks.diff.txt by zooko.

refs [#5399](https://github.com/twisted/twisted/issues/5399)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess set status to `closed`|
|-|-|

(In [33281]) Merge addcallbacks-ret-5399

Author: zooko
Reviewer: jesstess
Fixes: [#5399](https://github.com/twisted/twisted/issues/5399)

Document the return values of addCallbacks and chainDeferred in
twisted.internet.defer.Deferred.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess commented|
|-|-|

Thanks for the ticket and patch, zooko!

I added some [epytext](http://epydoc.sourceforge.net/epytext.html) markup before merging.

[Doc builder results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/addcallbacks-ret-5399).

(As usual, I failed to move this out of review before closing)
